Beta_Integrations/integration-Vertica.yml

```diff
@@ -0,0 +1,178 @@
+description: Analytic database management software
+configuration:
+- display: Host (Vertica Server URL)
+  name: url
+  defaultvalue: myhost.example.com
+  type: 0
+  required: true
+- display: Database
+  name: database
+  defaultvalue: mydb
+  type: 0
+  required: true
+- display: Username
+  name: credentials
+  defaultvalue: ""
+  type: 9
+  required: true
+- display: Port
+  name: port
+  defaultvalue: "5433"
+  type: 0
+  required: false
+script:
+  script: |-
+    ''' IMPORTS '''
+
+    import vertica_python
+    import datetime
+
+    ''' GLOBALS/PARAMS '''
+
+    USERNAME = demisto.params().get('credentials').get('identifier')
+    PASSWORD = demisto.params().get('credentials').get('password')
+    DATABASE = demisto.params().get('database')
+    PORT = int(demisto.params().get('port', 5433))
+    SERVER = demisto.params()['url'][:-1] if (demisto.params()['url'] and demisto.params()['url'].endswith('/')) else demisto.params()['url']
+    DB_PARAMS = {
+        'host': SERVER,
+        'port': PORT,
+        'user': USERNAME,
+        'password': PASSWORD,
+        'database': DATABASE,
+        'connection_timeout': 5
+    }
+
+    ''' HELPER FUNCTIONS '''
+
+    def convert_datetime_to_string(v):
+        """
+        Parses datetime object into string
+        """
+        if isinstance(v, datetime.datetime):
+            return v.strftime("%Y-%m-%dT%H:%M:%S")
+        return v
+
+
+    def connect_db():
+        try:
+            connection = vertica_python.connect(**DB_PARAMS)
+        except vertica_python.errors.ConnectionError:
+            return_error('Could not connect to DB, please re-check DB params.')
+        return connection
+
+
+    ''' COMMANDS + QUERY FUNCTIONS '''
+
+    def test_module():
+        """
+        Performs basic query on default system tables
+        """
+        cursor.execute("SELECT * FROM system_tables ORDER BY table_schema, table_name LIMIT 2;")
+        rows = cursor.fetchall()
+        if cursor.rowcount == 0:
+            return_error('No results were returned from the DB.')
+        demisto.results('ok')
+
+
+    def query_command():
+        """
+        Execute a query against the DB
+        """
+        # Init main vars
+        contents = []
+        context = {}
+        title = 'Vertica Query'
+
+        # Get arguments from user
+        query = demisto.args().get('query')
+        limit = int(demisto.args().get('limit', 50))
```

Comment:
not sure why we need this argument
if they would like to limit the results they can just add `LIMIT` to the query
wdyt ?

Reply:
Sort of fail safe in case someone is going to query something he is about to regret without considering it in advance. Which from my experience tends to happen.